utils/biospecimen.js

```diff
@@ -335,22 +315,34 @@ const biospecimenAPIs = async (req, res) => {
         if(req.method !== 'POST') {
             return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
         }
+        
         const requestData = req.body;
         if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
         
-        if(requestData['132929440']){
-            const boxId = requestData['132929440'];
-            const loginSite = requestData['789843387']
-            const { boxExists } = require('./firestore');
-            const exists = await boxExists(boxId, loginSite, requestData);
-            if (exists === false) return res.status(400).json(getResponseJSON('Box does not exist!', 400));
-            if (exists === true) {
-                const { updateBox } = require('./firestore');
-                await updateBox(boxId, requestData, loginSite);
-                return res.status(200).json({message: 'Success!', code:200})
-            }
-        } else {
-            return res.status(400).json({message: 'Error!', code:400});
+        try {
+            const { boxExists, updateBox } = require('./firestore');
+            
+            if (requestData[fieldMapping.shippingBoxId]){
+                const boxId = requestData[fieldMapping.shippingBoxId];
+                const loginSite = requestData[fieldMapping.loginSite];
+                const addedTubes = requestData['addedTubes'];
+        
+                if (!addedTubes || addedTubes.length === 0) return res.status(400).json(getResponseJSON('Missing added tubes', 400));
+
+                const exists = await boxExists(boxId, loginSite, requestData);
+                if (exists !== true) return res.status(400).json(getResponseJSON('Box does not exist!', 400));
```

Comment:
I am not sure if the front end cares or needs to differentiate easily between the invalid request above and the not found here but normally I would expect this to be a 404 error

Reply:
Good call. The front end doesn't care, but this is accurate. Updated. Thanks.